src/fsharp/FSharp.Core/prim-types.fsi

```diff
@@ -2740,6 +2746,44 @@ namespace Microsoft.FSharp.Core
         [<CompiledName("KeyValuePattern")>]
         val ( |KeyValue| ): keyValuePair:KeyValuePair<'Key,'Value> -> 'Key * 'Value
 
+        [<AutoOpen>]
+        module ArrayExtensions = 
```

Comment:
What are your thoughts on having new/different extensions for arrays related to slicing as type extensions themselves since the others (GetArraySlice/GetArraySlice2D/etc) are not? `GetReverseIndex` actually exists but `GetSlice` doesn't exist for arrays.

Reply:
If we wanted to wire up `GetReverseIndex` the same way we wire up `GetSlice`, we'd need to add 2 * (2^4 - 1) = 30 hardcoded slicing lookups to the big if statement in typechecker, and for Array4D alone. While we are introducing inconsistency, I feel like the way `GetSlice` is wired up should be the exception and not the norm, and there's little one can do shadowing `GetReverseIndex` only and not `GetSlice`.

Reply:
Awesome. Makes sense to me.